puppet/zulip_ops/manifests/postgres_common.pp

```diff
@@ -1,23 +1,14 @@
 class zulip_ops::postgres_common {
   include zulip::postgres_common
 
-  $internal_postgres_packages = [# dependencies for our wal-e backup system
-    'lzop',
-    'pv',
-    'python3-pip',
-    'python-pip',
-    'python3-gevent',
-    'python-gevent',
-  ]
-  package { $internal_postgres_packages: ensure => 'installed' }
+  exec {'wal-g-0.2.15':
+    command => 'cd /tmp && wget -qO wal-g.tar.gz https://github.com/wal-g/wal-g/releases/download/v0.2.15/wal-g.linux-amd64.tar.gz && tar xzf wal-g.tar.gz && mv wal-g /usr/local/bin/wal-g-0.2.15',
+    creates => '/usr/local/bin/wal-g-0.2.15',
```

Comment:
I'd like us to follow our standard practice of checking the hash of the package for third-party dependencies that we install not using an existing signed repository (like apt).

What do you think about extracting a script `scripts/setup/install-wal-g` that does the installation so that we can implement that logic not as a puppet one-liner?  We can access it via `"${::zulip_scripts_path}/setup/install-wal-g` in puppet, and this could just run that with a `creates => /usr/local/bin/wal-g`

I guess maybe that isn't right, because it wouldn't upgrade properly; so maybe the right answer is to have the script accept the version and hash as arguments, e.g. something like this:

```
$wal_g_version = "0.2.15"
$wal_g_version = "abcd1234"
 exec {'install_wal_g':
    command => '{::zulip_scripts_path}/setup/install-wal-g $wal_g_version $wal_g_hash',
    creates => '/usr/local/bin/wal-g-{wal_g_version}',
```

?

Reply:
I played briefly with doing this, but ISTR finding enough other places that installed unsigned packages that it wasn't worth it in this commit.

I'll see what makes sense to refactor out into a script based on other other usages in puppet; I think something more general may be possible.

Reply:
Ah, no -- it was that the previous technique (of just cloning the tip from git) wasn't pinned.  I've verified I don't see any other instances of installing random packages; I'll drop the wrapper into place as you suggest.